network-api/networkapi/wagtailpages/pagemodels/products.py

```diff
@@ -295,6 +310,45 @@ class ProductPage(FoundationMetadataPageMixin, Page):
         blank=True,
     )
 
+    # Un-editable voting fields. Don't add these to the content_panels.
+    current_total = models.IntegerField(default=0)
+    current_votecount = models.IntegerField(default=0)
+    vote_bin = models.OneToOneField(
+        ProductPageVoteBin,
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name='vote_bin',
+    )
+
+    @property
+    def average_vote_count(self):
+        try:
+            total = self.current_votecount / self.current_total
+        except ZeroDivisionError:
+            total = 0
+        return total
+
+    @property
+    def get_voting_json(self):
+        """
+        Return a dictionary as a string with the relevant data needed for the frontend:
+        """
+        json = {
+            'creepiness': {
+                'vote_breakdown': {
+                    '4': self.vote_bin.bin_5,
+                    '1': self.vote_bin.bin_4,
+                    '2': self.vote_bin.bin_3,
+                    '3': self.vote_bin.bin_2,
+                    '0': self.vote_bin.bin_1,
+                },
+                'average': self.average_vote_count
+            },
+            'total': self.current_votecount
+        }
+        return str(json)
```

Comment:
`str` does not yield json, and dicts don't serialize to json by default, so let's use the guaranteed correct function.
```suggestion
        return json.dumps(data)
```